internal/sync/diff.go

```diff
@@ -0,0 +1,226 @@
+				baseNormalizerPatch: baseNormalizerPatch{
+					groupKind: schema.GroupKind{Group: ignores[i].Group, Kind: ignores[i].Kind},
+					name:      ignores[i].Name,
+					namespace: ignores[i].Namespace,
+				},
+				patch: &patch,
+			})
+		}
+	}
+	return &ignoreNormalizer{patches: patches}, nil
+}
+
+// Normalize removes fields from supplied resource using json paths from matching items of specified resources ignored differences list
+func (n *ignoreNormalizer) Normalize(un *unstructured.Unstructured) error {
+	if un == nil {
+		return fmt.Errorf("invalid argument: unstructured is nil")
+	}
+	matched := make([]normalizerPatch, 0)
+	for _, patch := range n.patches {
+		groupKind := un.GroupVersionKind().GroupKind()
+
+		if glob.Match(patch.GetGroupKind().Group, groupKind.Group) &&
+			glob.Match(patch.GetGroupKind().Kind, groupKind.Kind) &&
+			(patch.GetName() == "" || patch.GetName() == un.GetName()) &&
+			(patch.GetNamespace() == "" || patch.GetNamespace() == un.GetNamespace()) {
+
+			matched = append(matched, patch)
+		}
+	}
+	if len(matched) == 0 {
+		return nil
+	}
+
+	docData, err := json.Marshal(un)
+	if err != nil {
+		return err
+	}
+
+	for _, patch := range matched {
+		patchedDocData, err := patch.Apply(docData)
+		if err != nil {
+			if shouldLogError(err) {
+				log.Debugf("Failed to apply normalization: %v", err)
+			}
+			continue
+		}
+		docData = patchedDocData
+	}
+
+	err = json.Unmarshal(docData, un)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func shouldLogError(e error) bool {
+	if strings.Contains(e.Error(), "Unable to remove nonexistent key") {
+		return false
+	}
+	if strings.Contains(e.Error(), "remove operation does not apply: doc is missing path") {
+		return false
+	}
+	return true
+}
+
+// StateDiffs will apply all required normalizations and calculate the diffs between
+// the live and the config/desired states.
+func StateDiffs(
+	configs, lives []*unstructured.Unstructured,
+	overrides map[string]ResourceOverride,
+	diffOpts []diff.Option) (*diff.DiffResultList, error) {
+	diffNormalizer, err := NewIgnoreNormalizer(overrides)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create diff normalizer: %w", err)
+	}
+	//
+	//resourceOps, cleanup, err := s.getResourceOperations()
+	//if err != nil {
+	//	log.Errorf("CompareAppState error getting resource operations: %s", err)
+	//}
+	//defer cleanup()
+
+	diffOpts = append(diffOpts, diff.WithNormalizer(diffNormalizer))
+	//diffOpts := []diff.Option{
+	//	//diff.WithLogr(),
+	//	diff.WithNormalizer(diffNormalizer),
+	//	//diff.IgnoreAggregatedRoles(diffConfig.IgnoreAggregatedRoles()),
+	//	//diff.WithStructuredMergeDiff(diffConfig.StructuredMergeDiff()),
+	//	//diff.WithGVKParser(diffConfig.GVKParser()),
+	//	//diff.WithManager(diffConfig.Manager()),
+	//	diff.WithServerSideDiff(true),
+	//	diff.WithServerSideDryRunner(diff.NewK8sServerSideDryRunner(resourceOps)),
+	//}
+	array, err := diff.DiffArray(configs, lives, diffOpts...)
```

Comment:
why is all of this commented out? does it need to be cleaned up? Or is it code we're considering putting back upon more investigation?

Reply:
sorry that needs to be cleaned up.